### PR TITLE
chore: Move playwright to optionalDependencies.

### LINKED ISSUE #475: Improve playwright harness
Currently:

1. it is not installing in any of my machines, the download get stuck forever
2. `npm install` get stuck and does not complete on CI (ref https://github.com/microsoft/playwright/issues/14853)

I think we should:

* [ ] split the integration / browser testing dependencies in its own `package.json` so we can avoid installing 113 MB
* [ ] ensure that the playwright "big download" is cached in CI so it is faster

@ShogunPanda could you take a look?

### PR DESCRIPTION
**Pull request by ShogunPanda**
Fixes #475 and #473.

### CODE REVIEW

**Comment by mcollina** [comment]
Can you put a timeouton the browser workflows?

**Comment by ShogunPanda** [comment]
@mcollina Done!

File: package.json [hunk 1]
```diff
@@ -77,6 +76,9 @@
     "webpack": "^5.72.1",
     "webpack-cli": "^4.9.2"
   },
+  "optionalDependencies": {
```
Comment (vweevers) [inline package.json:79]: optionalDependencies are production dependencies (i.e. will be installed by dependents)

**Comment by vweevers** [comment]
@mcollina I don't fully understand why you don't want to install playwright. Locally you'll only need to download the playwright binaries once, after that it's cached. Is it about CI? If so, you can selectively set `PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD=1`

**Comment by ShogunPanda** [comment]
@vweevers Yup, I think is to avoid to block CI in case those outages happen again. 

In general, since downloading browsers might be expensive (large files on mobile networks) I think we should not do it by default.

**Comment by mcollina** [comment]
Maybe this is a bad idea and we can just skip this? Everything seems to be actually working fine now.

**Comment by ShogunPanda** [comment]
Sounds good to me! Closing this!

**Pull request closed without merging**